### PR TITLE
Add ov5640 camera driver

### PR DESCRIPTION
**Pull request by ngphibang**
This PR adds ov5640 camera driver. It is splitted from [this ](https://github.com/zephyrproject-rtos/zephyr/pull/69810) to ease the review procerss.

It **already got reviewed** by Daniel Degrasse and others

### CODE REVIEW

File: soc/nxp/imxrt/imxrt11xx/soc.c [hunk 1]
```diff
@@ -398,6 +398,11 @@ static ALWAYS_INLINE void clock_init(void)
 	rootCfg.mux = kCLOCK_LPI2C5_ClockRoot_MuxOscRc48MDiv2;
 	rootCfg.div = 1;
 	CLOCK_SetRootClock(kCLOCK_Root_Lpi2c5, &rootCfg);
+
+	/* Configure Lpi2c6 using Osc24M */
```
Comment (danieldegrasse) [inline soc/nxp/imxrt/imxrt11xx/soc.c:402]: This really should be in a separate PR from this one- it is clearly something we are missing on the RT1170 clock setup, but this PR is just enabling the camera driver

**Review by danieldegrasse — COMMENTED** [review]
We should add an entry for this camera driver to the application devicetree overlay in `tests/drivers/build_all/video`, so that this driver gets built within CI. Also- please take a look at CI failures when you get a chance:)

**Comment by josuah** [comment]
Thank you for adding this driver!

Here are a few complementary sources for anyone interested in this sensor, more or less obvious:
- https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/media/i2c/ov5640.c
- https://github.com/adafruit/Adafruit_CircuitPython_OV5640/blob/main/adafruit_ov5640.py
- https://github.com/STMicroelectronics/stm32-ov5640/blob/main/ov5640.c
- https://github.com/surgical-robots/cycx3_uvc_ov5640/blob/master/cy_ov5640.c

Some might use extra information from the vendors not available in the [datasheet](https://cdn.sparkfun.com/datasheets/Sensors/LightImaging/OV5640_datasheet.pdf), which is great but not really self-sufficient.

**Review by loicpoulain — DISMISSED** [review]
LGTM for ov5640.

**Comment by ngphibang** [comment]
@danieldegrasse : Is it ok for you now ?

**Comment by ngphibang** [comment]
@loicpoulain : I force pushed to address @danieldegrasse 's comments as well as to resolve the merge conflict (there's a DCMI camera driver merged meanwhile). Could you please re-approve the PR ?

**Review by danieldegrasse — APPROVED** [review]

**Comment by ngphibang** [comment]
@galak Could you help approve and merge this PR, please ? It got 2 approvals but one is cleared due to forced push.
Thanks !

**Comment by ngphibang** [comment]
Please, could someone re-add @loicpoulain's approval so that this can be merged rapidly as this is a blocking point for other PRs ?

**Review by dleach02 — APPROVED** [review]

**Review by loicpoulain — APPROVED** [review]

**Comment by dleach02** [comment]
@decsny can you review as a co-assignee. The bot assigned @galak this because of the dts change.

**Comment by decsny** [comment]
dts change is trivial, reassigning to loic because I don't know anything about video
